cmd/opampsupervisor/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/opampsupervisor
 
-go 1.24.6
+go 1.24.9
```

Comment:
nit: Do we need to update the go mods? Shouldn't they pick up whatever version of go is provided so long as it's greater than the minimum version in the go.mod? I would think we'd just need to update the workflows.